packages/block-editor/src/components/link-control/index.js

```diff
@@ -376,9 +375,9 @@ function LinkControl( {
 				/>
 			) }
 
-			{ isEditing && (
+			{ isEditingLink && hasLinkValue && (
 				<div className="block-editor-link-control__tools">
-					{ showSettings && (
```

Comment:
We cannot remove this check for `showSettings`.

Reply:
We need isLinkEditing and hasLinkValue as well, so the additional controls and settings don’t render on the initial state, before you’ve selected a link. How do you suggest we do both, wrap this in showSettings?

Reply:
`showSettings` is a const var which should be relied upon to determine the state under which the settings controls should be rendered.

Therefore I think the conditions you list should become part of that variable.

Reply:
Added in https://github.com/WordPress/gutenberg/pull/51712/commits/0408cbf170c322a38fce1e67acc08e30966894d7